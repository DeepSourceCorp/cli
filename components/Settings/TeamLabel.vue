<template>
  <nuxt-link
    :to="settingsLink"
    class="rounded-sm bg-vanilla-100 bg-opacity-10 px-1 text-cherry-150"
    >{{ login }}</nuxt-link
  >
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'

/**
 * A clickable label for a team. Links to the members settings page.
 */
@Component({})
export default class TeamLabel extends Vue {
  @Prop()
  login: string

  @Prop()
  vcsProvider: string

  get settingsLink() {
    return `/${this.vcsProvider.toLowerCase()}/${this.login}/members/active`
  }
}
</script>
