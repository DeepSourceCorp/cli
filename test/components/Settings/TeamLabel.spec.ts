import TeamLabel from '~/components/Settings/TeamLabel.vue'
import { render } from '@testing-library/vue'

describe('[[ TeamLabel ]]', () => {
  test('renders TeamLabel with all props', () => {
    const { html } = render(TeamLabel, {
      props: {
        login: 'deepsourcelabs',
        vcsProvider: 'GH'
      }
    })

    expect(html()).toMatchSnapshot()
  })
})
