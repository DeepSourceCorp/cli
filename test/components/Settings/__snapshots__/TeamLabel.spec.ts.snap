// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[[ TeamLabel ]] renders TeamLabel with all props 1`] = `"<nuxt-link to=\\"/gh/deepsourcelabs/members/active\\" class=\\"bg-vanilla-100 bg-opacity-10 px-1 rounded-sm text-cherry-150\\">deepsourcelabs</nuxt-link>"`;
